rust/operator-binary/src/controller.rs

```diff
@@ -475,6 +553,23 @@ fn build_rolegroup_statefulset(
         env.push(hive);
     };
 
+    if let Some(S3ConnectionSpec {
+        secret_class: Some(secret_name),
+        ..
+    }) = s3_connection_spec
+    {
+        env.push(env_var_from_secret(
+            ENV_S3_ACCESS_KEY,
+            secret_name,
+            SECRET_KEY_S3_ACCESS_KEY,
+        ));
+        env.push(env_var_from_secret(
+            ENV_S3_SECRET_KEY,
+            secret_name,
+            SECRET_KEY_S3_SECRET_KEY,
+        ));
```

Comment:
The `ContainerBuilder` from operator-rs has a method `add_env_var_from_source` which would make `env_var_from_secret` obsolete.

Reply:
Yes but the pattern in this function is to collect all environment variables first and then add all of them at once to the container builder (see `.add_env_vars(env)`). I actually tried the `add_env_var_from_secret` function of the builder but the code looked cleaner this way. But that is a matter of taste. Feel free to change it.